### ISSUE TITLE
Facebook Account Error

### ISSUE THREAD

**Issue by masonry47** [engram-design/SocialPoster#39]
### Describe the bug

Hey there again!  

Thanks for the previous bug fixes. We have something new in trying to get this setup.  Not sure if it is a bug but now saving the account is successful after the update, but when trying to "connect" that account we are getting an error on the Facebook side.

The URL thats loads is https://www.facebook.com/v3.0/dialog/oauth?scope=publish_to_groups%2Cuser_posts%2Cuser_photos%2Cmanage_pages%2Cpublish_pages&state=e6306eaf9dd0cd00846ab8dee6d3b74a&response_type=code&approval_prompt=auto&redirect_uri=https%3A%2F%2test.ddev.site%3A8443%2Findex.php%2Factions%2Fsocial-poster%2Faccounts%2Fcallback&client_id=645439323236332

The response is just a generic:

"Sorry, something went wrong.
We're working on getting this fixed as soon as we can."

But it doesn't really seem to be just a temporary thing or a caching or some sort of session issue.

Let us know if you have any ideas and thanks again!

### Steps to reproduce

1. Configure Facebook Provider
2. Create a Facebook Page account associated with this Provider
3. Try to "connect" this account

### Craft CMS version

4.2.5.2

### Plugin version

3.0.1

### Multi-site?

No

**Comment by engram-design** [comment]
Are you able to look in your `storage/logs/social-poster.log` for more information on the issue?

Can you also confirm your Redirect URI? There are some limitations as stated in the [docs](https://verbb.io/craft-plugins/social-poster/docs/providers/facebook#error-incorrect-redirect-url)

**Comment by masonry47** [comment]
Nothing gets logged in there.   And just to clarify that response above shows on the Facebook URL. We never even get returned to Craft after hitting "connect".

On the redirect URL, we are using the usePathInfo config so the redirect URI looks like this 

https://test.ddev.site:8443/index.php/actions/social-poster/accounts/callback

Thanks!

**Comment by engram-design** [comment]
Ah, looks like I hadn't merged the changes from the Craft 3 version of Social Poster. We weren't supporting the latest Facebook API, which would be the cause for this issue.

Fixed for the next release. To get the fix early, change your `verbb/social-poster` requirement in `composer.json` to:

```
"require": {
  "verbb/social-poster": "dev-craft-4 as 3.0.1",
  "...": "..."
}
```

Then run `composer update`.

In addition, I've also changed the Redirect URI to prevent any issues with `usePathInfo`. You can use both, but you might fare better with the newer URI (just refresh your provider settings screen).

**Comment by masonry47** [comment]
Thanks!  That got us connected.  When actual trying to post we now get a new response:

`Could not post to Facebook. Error: {"statusCode":"[error]","reasonPhrase":"(#100) Tried accessing nonexisting field (access_token) on node type (Business)"}.`

May be a config thing on our end but thought we would check.

Also one question, we see in the docs what the review process is for Facebook.  Are we able to go ahead and post though without that if the app in Facebook is in "Development" mode?  From what we understand the app should be fully built and working before submitting for review.  Thanks!

**Comment by engram-design** [comment]
I've not seen that error before. Are you able to look in your `storage/logs/social-poster.log` to see if there's more detail?

As for the Facebook app approval process, you'll need to set your app to "Live" in order to have the posts visible publicly. To have it "Live" you'll need to apply for access to some of the permissions to post publicly to Facebook. Keeping the app in development mode will work and you'll see your posts on Facebook, but only you will be able to see your posts from Craft (they won't be public).

**Comment by masonry47** [comment]
Thanks for looking into this.  This only thing additional is the log is this

`2022-10-17 06:57:21 [ERROR] Error posting to Facebook: “(#100) Tried accessing nonexisting field (access_token) on node type (Business)” /var/www/html/vendor/janu-software/facebook-php-sdk/src/Exception/ResponseException.php:72`

Thanks!

**Comment by masonry47** [comment]
@engram-design Just following up on this.  Really appreciate it!

**Comment by engram-design** [comment]
Is this a Page or Group you're posting to? I can't seem to replicate this error, but it looks like this is an issue fetching the [access token](https://github.com/verbb/social-poster/blob/3bdbec3ad307c547bf3f09f0754e29dcd44daaf6/src/providers/Facebook.php#L116) from Facebook.

Can you tell me the state of your Facebook app? Whether it's in live or development mode, if you've applied for the extra permissions, if you have admin access to the page/group for the account that authorized the login.

Are you posting to a Business page as opposed to a Personal page?

**Comment by engram-design** [comment]
In addition, I've also made some changes along the way, if you can run `composer update` before trying again.

**Comment by masonry47** [comment]
Thanks!  Looks like these changes got us going.  Really appreciate it.

**Comment by engram-design** [comment]
Should be fixed in [3.0.2](https://github.com/verbb/social-poster/releases/tag/3.0.2)
